lib/generators/doorkeeper/previous_refresh_token_generator.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails/generators/active_record'
+
+module Doorkeeper
+  class PreviousRefreshTokenGenerator < Rails::Generators::Base
+    include Rails::Generators::Migration
+    source_root File.expand_path('../templates', __FILE__)
+    desc 'Support revoke refresh token on access token use'
+
+    def self.next_migration_number(path)
+      ActiveRecord::Generators::Base.next_migration_number(path)
+    end
+
+    def previous_refresh_token
+      if !previous_refresh_token_column_exists?
```

Comment:
> Favor `unless` over `if` for negative conditions.

Ir we can say `if no_previous_refresh_token_column?`, which I think would read best.